hibernate-core/src/main/java/org/hibernate/engine/internal/Nullability.java

```diff
@@ -95,10 +94,16 @@ public void checkNullability(
 					? persister.getPropertyInsertability()
 					: persister.getPropertyUpdateability();
 			final Type[] propertyTypes = persister.getPropertyTypes();
+			final InMemoryValueGenerationStrategy[] inMemoryValueGenerationStrategies =
+					persister.getEntityMetamodel().getInMemoryValueGenerationStrategies();
+
+			boolean validValueGenerationStrategyCount = inMemoryValueGenerationStrategies.length == values.length;
 
 			for ( int i = 0; i < values.length; i++ ) {
 
-				if ( checkability[i] && values[i]!= LazyPropertyInitializer.UNFETCHED_PROPERTY ) {
+				if ( checkability[i] && values[i] != LazyPropertyInitializer.UNFETCHED_PROPERTY &&
+						( validValueGenerationStrategyCount && GenerationTiming.NEVER == inMemoryValueGenerationStrategies[i]
```

Comment:
Are you sure you want this expression to return `false` if `! validValueGenerationStrategyCount `?

Otherwise:
```suggestion
						( !validValueGenerationStrategyCount || GenerationTiming.NEVER == inMemoryValueGenerationStrategies[i]
						
```

What kind of an odd situation by the way, to have `inMemoryValueGenerationStrategies.length == values.length`...

Reply:
In fact, we don't need the `validValueGenerationStrategyCount` at all. The problem was the `CustomPersister` using the wrong `Custom.hbm.xml``mapping.

Reply:
Cool! Thanks for verifying this!